src/Routing/Route/Route.php

```diff
@@ -444,7 +439,11 @@ public function parse(string $url, string $method): ?array
         // Assign defaults, set passed args to pass
         foreach ($this->defaults as $key => $value) {
             if (isset($route[$key])) {
-                continue;
+                // preg_match returns an empty string if an optional key was not fournd
+                // We treat this as if the key was not found
+                if ($route[$key] !== '') {
```

Comment:
Couldn't this condition be combined with the one above?

Reply:
Done.